### PR TITLE
Added use_literal_group_id argument to artifactory.downloaded

### PR DESCRIPTION
**Pull request by calebwoofenden**
### What does this PR do?
This adds an argument to the artifactory.downloaded module that allows the user to specify whether or not to use the literal group_id when building the artifactory url to download from.

This argument defaults to false and is optional, so if not provided behavior won't change at all. If true, dots in provided group_id will not be replaced with slashes when building the artifactory url.

### What issues does this PR fix or reference?
It addresses this issue: https://github.com/saltstack/salt/issues/42510

### Previous Behavior
If a user provided a group_id that contained dots, e.g. com.company, it would change this to com/company when building the URL to download from. If the URL actually contained com.company, this would result in an incorrect URL and failed download.

### New Behavior
Now the user can add an argument to artifactory.downloaded to tell it to keep the dots in the provided group_id.

### Tests written?

No

### CODE REVIEW

File: salt/modules/artifactory.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def __virtual__():
         return True
 
 
-def get_latest_snapshot(artifactory_url, repository, group_id, artifact_id, packaging, target_dir='/tmp', target_file=None, classifier=None, username=None, password=None):
+def get_latest_snapshot(artifactory_url, repository, group_id, artifact_id, packaging, target_dir='/tmp', target_file=None, classifier=None, username=None, password=None, use_literal_group_id=False):
```
Comment (cachedout) [inline salt/modules/artifactory.py]: Is this expected to work on Windows? If so, I'm wondering if `target_dir` should be `None` and then we transform it in the implementation? If not, we should specifically exclude Windows in `__virtual__`.

I know that's not part of your change but I just saw it. :]

File: salt/modules/artifactory.py [hunk 2]
```diff
@@ -465,7 +468,7 @@ def __save_artifact(artifact_url, target_file, headers):
     try:
         request = urllib.request.Request(artifact_url, None, headers)
         f = urllib.request.urlopen(request)
-        with salt.utils.files.fopen(target_file, "wb") as local_file:
+        with salt.utils.fopen(target_file, "wb") as local_file:
```
Comment (cachedout) [inline salt/modules/artifactory.py]: Why change this back? `salt.utils.fopen` is the preference here and `salt.utils.fopen` is being deprecated.

Comment (calebwoofenden) [inline salt/modules/artifactory.py]: Good question, I did not change that. I'm not sure how it got changed!

File: salt/modules/artifactory.py (hunk 1 above)
Comment (calebwoofenden) [inline salt/modules/artifactory.py]: Yes, we are currently trying to use this for a Windows 2008 R2 minion and I've also been testing on a Windows 7 machine. It seems to work fine as is, though I don't think our use case actually uses that function.

Comment (cachedout) [inline salt/modules/artifactory.py]: K. I'll merge this as-is but a follow-up PR for this might be good. I will see if I can find the time to put one in.

**Pull request merged**